src/platform/unix/mod.rs

```diff
@@ -873,50 +922,37 @@ fn recv(fd: c_int, blocking_mode: BlockingMode)
     Ok((main_data_buffer, channels, shared_memory_regions))
 }
 
-#[cfg(target_os="android")]
-fn maybe_unlink(_: *const c_char) -> c_int {
-    // Calling `unlink` on a file stored on an sdcard immediately deletes it.
-    // FIXME: use a better temporary directory than the sdcard via the Java APIs
-    // and threading that value into Servo.
-    // https://code.google.com/p/android/issues/detail?id=19017
-    0
+fn create_string() -> CString {
+    let epoc = UNIX_EPOCH.elapsed().unwrap_or(Duration::new(0, 0));
+    let pid = unsafe {
+        libc::getpid()
+    };
+    CString::new(&*format!("/ipc-channel-shared-memory.{}.{}.{}",
+                           pid, epoc.as_secs(), epoc.subsec_nanos())).unwrap()
 }
 
-#[cfg(not(target_os="android"))]
-unsafe fn maybe_unlink(c: *const c_char) -> c_int {
-    libc::unlink(c)
+#[cfg(any(feature="force-posix-shm", not(any(target_os="linux", target_os="android"))))]
+unsafe fn create_shmem(name: *const c_char, length: usize) -> c_int {
+    let fd = libc::shm_open(name, libc::O_CREAT | libc::O_RDWR | libc::O_EXCL, 0o600);
```

Comment:
A quick check seems to confirm that this works with `0o000` as well, as I expected :-)